rest/routing.go

```diff
@@ -136,160 +136,159 @@ func CreateAdminRouter(sc *ServerContext) *mux.Router {
-		makeHandler(sc, adminPrivs, nil, nil, (*handler).getUserSession)).Methods("GET")
+		makeHandler(sc, adminPrivs, []Permission{PermReadPrincipal}, nil, (*handler).getUserSession)).Methods("GET")
 
 	dbr.Handle("/_session/{sessionid}",
-		makeHandler(sc, adminPrivs, nil, nil, (*handler).deleteUserSession)).Methods("DELETE")
+		makeHandler(sc, adminPrivs, []Permission{PermWritePrincipal}, nil, (*handler).deleteUserSession)).Methods("DELETE")
 
 	dbr.Handle("/_raw/{docid:"+docRegex+"}",
-		makeHandler(sc, adminPrivs, nil, nil, (*handler).handleGetRawDoc)).Methods("GET", "HEAD")
+		makeHandler(sc, adminPrivs, []Permission{PermReadAppData}, nil, (*handler).handleGetRawDoc)).Methods("GET", "HEAD")
 
 	dbr.Handle("/_revtree/{docid:"+docRegex+"}",
-		makeHandler(sc, adminPrivs, nil, nil, (*handler).handleGetRevTree)).Methods("GET")
+		makeHandler(sc, adminPrivs, []Permission{PermReadAppData}, nil, (*handler).handleGetRevTree)).Methods("GET")
 
 	dbr.Handle("/_user/",
-		makeHandler(sc, adminPrivs, nil, nil, (*handler).getUsers)).Methods("GET", "HEAD")
+		makeHandler(sc, adminPrivs, []Permission{PermReadPrincipal}, []Permission{PermReadPrincipal}, (*handler).getUsers)).Methods("GET", "HEAD")
 	dbr.Handle("/_user/",
-		makeHandler(sc, adminPrivs, nil, nil, (*handler).putUser)).Methods("POST")
+		makeHandler(sc, adminPrivs, []Permission{PermWritePrincipal}, nil, (*handler).putUser)).Methods("POST")
 	dbr.Handle("/_user/{name}",
-		makeHandler(sc, adminPrivs, nil, nil, (*handler).getUserInfo)).Methods("GET", "HEAD")
+		makeHandler(sc, adminPrivs, []Permission{PermReadPrincipal, PermReadPrincipalAppData}, []Permission{PermReadPrincipal, PermReadPrincipalAppData}, (*handler).getUserInfo)).Methods("GET", "HEAD")
 	dbr.Handle("/_user/{name}",
-		makeHandler(sc, adminPrivs, nil, nil, (*handler).putUser)).Methods("PUT")
+		makeHandler(sc, adminPrivs, []Permission{PermWritePrincipal}, nil, (*handler).putUser)).Methods("PUT")
 	dbr.Handle("/_user/{name}",
-		makeHandler(sc, adminPrivs, nil, nil, (*handler).deleteUser)).Methods("DELETE")
+		makeHandler(sc, adminPrivs, []Permission{PermWritePrincipal}, nil, (*handler).deleteUser)).Methods("DELETE")
 
 	dbr.Handle("/_user/{name}/_session",
-		makeHandler(sc, adminPrivs, nil, nil, (*handler).deleteUserSessions)).Methods("DELETE")
+		makeHandler(sc, adminPrivs, []Permission{PermWritePrincipal}, nil, (*handler).deleteUserSessions)).Methods("DELETE")
 	dbr.Handle("/_user/{name}/_session/{sessionid}",
-		makeHandler(sc, adminPrivs, nil, nil, (*handler).deleteUserSession)).Methods("DELETE")
+		makeHandler(sc, adminPrivs, []Permission{PermWritePrincipal}, nil, (*handler).deleteUserSession)).Methods("DELETE")
 
 	dbr.Handle("/_role/",
-		makeHandler(sc, adminPrivs, nil, nil, (*handler).getRoles)).Methods("GET", "HEAD")
+		makeHandler(sc, adminPrivs, []Permission{PermReadPrincipal, PermReadPrincipalAppData}, []Permission{PermReadPrincipal, PermReadPrincipalAppData}, (*handler).getRoles)).Methods("GET", "HEAD")
```

Comment:
I expected this to be just readPrincipal, similar to /_user/, as it's also just returning a list of IDs.

Reply:
Yup this was my mistake.